curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67f8ec758a740e34432222db.md

```diff
@@ -0,0 +1,114 @@
+---
+id: 67f8ec758a740e34432222db
+title: Step 8
+challengeType: 0
+dashedName: step-8
+---
+
+# --description--
+
+Add an `onChange` handler to the `input` element that updates the `query` state.
+
+# --hints--
+
+You should ensure that the `onChange` handler is defined for the `input` element, and it updates the `query` state properly.
+
+```js
+assert.match(code, /onChange=\{\(?e\)?\s*=>\s*setQuery\(e\.target\.value\)\}/)
```

Comment:
This regex seems a bit strict. What if I use anything else in place of `e`?

Reply:
I would suggest looking at the color picker lab for alternates in testing. that lab has more flexibility in allowing for correct answers.